src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAsyncIteratorTests.cs

```diff
@@ -5012,5 +5012,782 @@ public async System.Collections.Generic.IAsyncEnumerator<int> GetAsyncEnumerator
+        public void CancellationTokenParameter_Overridding()
+        {
+            string source = @"
+using static System.Console;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+public class Base
+{
+    public virtual async System.Collections.Generic.IAsyncEnumerable<int> Iter([DefaultCancellation] CancellationToken token1, int value)
+    {
+        yield return value++;
+        await Task.Yield();
+    }
+}
+public class C : Base
+{
+    static async Task Main()
+    {
+        var enumerable = new C().Iter(default, 42);
+
+        using CancellationTokenSource source = new CancellationTokenSource();
+        CancellationToken token = source.Token;
+        source.Cancel();
+        await using var enumerator = enumerable.GetAsyncEnumerator(token); // some token passed but unconsumed 
+
+        if (!await enumerator.MoveNextAsync()) throw null;
+        System.Console.Write($""{enumerator.Current}""); // 42
+    }
+    public override async System.Collections.Generic.IAsyncEnumerable<int> Iter(CancellationToken token1, int value)
+    {
+        await Task.Yield();
+        token1.ThrowIfCancellationRequested();
+        Write(""Reached "");
+        yield return value;
+    }
+}";
+            // The overridden method lacks the DefaultCancellation attribute
+            var comp = CreateCompilationWithAsyncIterator(new[] { source, DefaultCancellationAttributeType }, options: TestOptions.DebugExe);
+            comp.VerifyDiagnostics();
+            CompileAndVerify(comp, expectedOutput: "Reached 42");
+        }
+
+        [Fact, WorkItem(34407, "https://github.com/dotnet/roslyn/issues/34407")]
+        public void CancellationTokenParameter_Overridding2()
+        {
+            string source = @"
+using static System.Console;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+public class Base
+{
+    public virtual async System.Collections.Generic.IAsyncEnumerable<int> Iter(CancellationToken token1, int value)
+    {
+        yield return value++;
+        await Task.Yield();
+    }
+}
+public class C : Base
+{
+    static async Task Main()
+    {
+        var enumerable = new C().Iter(default, 42);
+
+        using CancellationTokenSource source = new CancellationTokenSource();
+        CancellationToken token = source.Token;
+        await using var enumerator = enumerable.GetAsyncEnumerator(token); // some token passed
+
+        if (!await enumerator.MoveNextAsync()) throw null;
+        System.Console.Write($""{enumerator.Current} ""); // 42
+
+        source.Cancel();
+        try
+        {
+            await enumerator.MoveNextAsync();
+        }
+        catch (System.OperationCanceledException)
+        {
+            Write(""Cancelled"");
+        }
+    }
+    public override async System.Collections.Generic.IAsyncEnumerable<int> Iter([DefaultCancellation] CancellationToken token1, int value)
```

Comment:
Do you have any tests where the iterator `DefaultCancellation` token parameter has a default value?

Reply:
I figured that was orthogonal, but I can add one.